form-design.rst

```diff
@@ -0,0 +1,420 @@
+****************
+Form Design
```

Comment:
XForm, not just Form

Is this about designing forms? Is it about coding forms?

Reply:
This is exactly the anatomy of XForm, I have changed the name to that now. Is it okay?